pkg/controller/istiocontrolplane/listeners.go

```diff
@@ -286,31 +287,38 @@ func (c *CitadelChartCustomizer) cleanCitadelResources(obj runtime.Object) error
 		return fmt.Errorf("get error to get namespace of %s: %s", obj.GetObjectKind().GroupVersionKind(), err)
 	}
 
-	client := c.Reconciler.GetClient()
+	rClient := c.Reconciler.GetClient()
 	cmNamespace := objAccessor.GetNamespace()
 	cmName := "istio-security"
-	prunedSecretType := "istio.io/key-and-cert"
-
+	prunedSecretTypeMap := map[string]bool{
+		"istio.io/key-and-cert": true,
+		"istio.io/ca-root":      true,
```

Comment:
Also add secret with type `istio.io/dns-key-and-cert`,  per @linsun 's comments: https://github.com/istio/operator/pull/415/files#r336817378

Others look good to me.

Reply:
added.